src/mixxx.cpp

```diff
@@ -708,65 +700,125 @@ int MixxxMainWindow::noSoundDlg(void) {
         } else if (msgBox.clickedButton() == reconfigureButton) {
             msgBox.hide();
 
+            m_pSoundManager->clearAndQueryDevices();
             // This way of opening the dialog allows us to use it synchronously
             m_pPrefDlg->setWindowModality(Qt::ApplicationModal);
             m_pPrefDlg->exec();
             if (m_pPrefDlg->result() == QDialog::Accepted) {
-                return 0;
+                return QDialog::Accepted;
             }
 
             msgBox.show();
-
         } else if (msgBox.clickedButton() == exitButton) {
-            return 1;
+            // Will finally quit Mixxx
+            return QDialog::Rejected;
         }
     }
 }
 
-int MixxxMainWindow::noOutputDlg(bool *continueClicked) {
+QDialog::DialogCode MixxxMainWindow::soundDeviceBusyDlg(bool* retryClicked) {
+    QString title(tr("Sound Device Busy"));
+    QString text(
+            "<html> <p>" %
+            tr("Mixxx was unable to open all the configured sound devices.") +
+            "</p> <p>" %
+            m_pSoundManager->getErrorDeviceName() %
+            " is used by an other application or not plugged in."
```

Comment:
It would be nice to report which case it is (in use by another application or unplugged), but if that's too much trouble to implement, don't worry about it.